File: PakClassified.WebApp/PakClassified.WebApp.DTOs/PakClassified.DTOs/AdvertisementSearchFilterModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PakClassified.WebApp.DTOs.PakClassified.DTOs
{
    public class AdvertisementSearchFilterModel
    {
        public int? CityAreaId { get; set; }
        public int? PostedById { get; set; }
        public int? StatusId { get; set; }
        public int? TypeId { get; set; }
        public int? SubCategoryId { get; set; }
        public ICollection<int>? TagIds { get; set; }
    }
}

File: PakClassified.Client/src/app/core/models/pakClassified/advertisement-search-filter-model.ts
export interface AdvertisementSearchFilter {
    cityAreaId?: number;
    postedById?: number;
    statusId?: number;
    typeId?: number;
    subCategoryId?: number;
    tagIds?: number[];
}

export interface SearchFilterModal extends AdvertisementSearchFilter {
    cityArea?: string;
    postedBy?: string;
    status?: string;
    type?: string;
    subCategory?: string;
    tagNames?: string[];
}